block_iagora.php

```diff
@@ -48,18 +48,31 @@ public function get_content() {
 
         $this->content = new stdClass();
         $this->content->footer = '';
+        $copilotendpointurl = isset($this->config->copilotendpointurl) ? $this->config->copilotendpointurl : '';
 
-        $iframeurl = isset($this->config->iframeurl) ? $this->config->iframeurl : '';
-
-        if (empty($iframeurl)) {
-            $this->content->text = get_string('noiframeurl', 'block_iagora');
+        if (empty($copilotendpointurl)) {
+            $this->content->text = get_string('nocopilotendpointurl', 'block_iagora');
         } else {
-            $this->content->text = '<iframe src="' . $iframeurl . '" width="100%" height="400px" frameborder="0"></iframe>';
+            $this->content->text = $this->generate_chat_content($copilotendpointurl);
         }
-
         return $this->content;
     }
-
+    /**
+     * Generate the HTML content for the chat block.
+     *
+     * @param string $copilotendpointurl The URL to be used in the iframe.
+     * @return string The generated HTML content for the chat.
+     */
+    private function generate_chat_content($copilotendpointurl) {
+        global $OUTPUT;
+        // Generate a unique identifier for the chat container.
+        $chatid = uniqid('iagora_chat_');
+        $context = [
+            'chatid' => $chatid,
```

Comment:
```suggestion
            'chatId' => $chatid,
```
See [moodle JavaScript variables naming convention](https://docs.moodle.org/dev/Javascript/Coding_Style#Variable_and_function_naming).